native/ex_tokenizers/src/decoders.rs

```diff
@@ -0,0 +1,233 @@
+use std::collections::HashMap;
+
+use tokenizers::{Decoder, DecoderWrapper};
+
+use crate::{util::DetailValue, ExTokenizersError};
+
+pub struct ExTokenizersDecoderRef(pub DecoderWrapper);
+
+#[derive(rustler::NifStruct)]
+#[module = "Tokenizers.Decoder"]
+pub struct ExTokenizersDecoder {
+    pub resource: rustler::resource::ResourceArc<ExTokenizersDecoderRef>,
+}
+
+impl ExTokenizersDecoderRef {
+    pub fn new<T>(data: T) -> Self
+    where
+        T: Into<DecoderWrapper>,
+    {
+        Self(data.into())
+    }
+}
+
+impl ExTokenizersDecoder {
+    pub fn new<T>(data: T) -> Self
+    where
+        T: Into<DecoderWrapper>,
+    {
+        Self {
+            resource: rustler::resource::ResourceArc::new(ExTokenizersDecoderRef::new(data)),
+        }
+    }
+}
+
+#[rustler::nif(schedule = "DirtyCpu")]
+fn decoders_decode(
+    decoder: ExTokenizersDecoder,
+    tokens: Vec<String>,
+) -> Result<String, ExTokenizersError> {
+    decoder
+        .resource
+        .0
+        .decode(tokens)
+        .map_err(|e| ExTokenizersError::Tokenizer(e))
+}
+
+///////////////////////////////////////////////////////////////////////////////
+/// Inspection
+///////////////////////////////////////////////////////////////////////////////
+
+#[rustler::nif]
+fn decoders_info(
+    decoder: ExTokenizersDecoder,
+) -> Result<HashMap<String, DetailValue>, ExTokenizersError> {
```

Comment:
I think we may want to optimize this result to return a Term representing the Map. It's not needed for this PR, but just to keep an eye on it.

Reply:
I haven't got it. Can you please share a short snippet mb?